### ISSUE TITLE
Support linting .d.ts files with `--project` flag

### ISSUE THREAD

**Issue by spalger** [palantir/tslint#3966]
### Feature Request

- __TSLint version__: ^5.10.0
- __TypeScript version__: ^2.8.3
- __Running TSLint via__: CLI

Our pre-commit hook lints all modified files with `.ts` extensions with tslint, but when we are linting all projects in our repo with call tslint multiple times with the `--project` flag. I noticed today that when I modified a `.d.ts` file it had linting errors that we not found when using `--project` because tslint explicitly ignores them in [`Linter.getFileNames()`](https://github.com/palantir/tslint/blob/46098f7b7f0f63f8d04b63e9a3598d1fc467b35f/src/linter.ts#L94-L102).

We want to lint our definition files, and I think we should be able to even when using the `--project` flag. Do you think I could submit a PR that adds a `--include-project-definitions` flag, or maybe one that enables them by default? The TSLint plugin for VSCode supports the flag `tslint.ignoreDefinitionFiles` which is disabled by default, and I'm not sure how long `--project` has been in wide use to select files for linting, but I was quite surprised by this behavior and suspect others might be too.

**Comment by giladgray** [comment]
@spalger sure we'll take a look at a PR. not sure how much work this'll require. good luck!

**Comment by kkirkfield** [comment]
@spalger Did you ever get a chance to look at this? If not I may look into it.
For some of my TS projects I use Gulp and pass the files to TSLint using the include property in the tsconfig.json. *.d.ts files don't seem to be supported. Adding support for including *.d.ts files would mean only needing to define file globs in one place.
```javascript
function lint() {
	return new Promise((resolve, reject) => {
		var program = tslint.Linter.createProgram('tsconfig.json');
		var files = tslint.Linter.getFileNames(program);
		gulp.src(files)
			.pipe(gulpTslint({
				configuration: 'tslint.json',
				fix: argv.fix,
				formatter: 'verbose',
				program
			}))
			.on('error', reject)
			.pipe(gulpTslint.report({
				allowWarnings: true,
				summarizeFailureOutput: true
			}))
			.on('error', reject)
			.on('end', resolve);
	});
}
```

**Comment by spalger** [comment]
I've looked at it a couple times but it's not an urgent need for our project, so I haven't really prioritized it. If you'd like to give it a shot @kkirkfield be my guest! I'd happily checkout the PR and make sure it works for me too.

**Comment by massimonewsuk** [comment]
I think my issue is related to this: https://github.com/palantir/tslint/issues/3922

**Comment by VincentLanglet** [comment]
Any update ?

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for major new changes or features. See #4534. 😱

If you'd like to see this change implemented, you have two choices:
* **Recommended:** Check if this is available in [ESLint](https://eslint.org) + [typescript-eslint](https://typescript-eslint.io) ✅
* _Not Recommended: Fork TSLint locally_ 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
